tests/unit/playwright_crawler/test_playwright_crawler.py

```diff
@@ -133,6 +133,24 @@ async def request_handler(context: PlaywrightCrawlingContext) -> None:
     assert headers['User-Agent'] == PW_FIREFOX_HEADLESS_DEFAULT_USER_AGENT
 
 
+async def test_custom_headers() -> None:
+    crawler = PlaywrightCrawler()
+    headers = dict[str, str]()
+    custom_headers = {'Power-Header': 'ring', 'Library': 'storm', 'My-Test-Header': 'fuzz'}
+
+    @crawler.router.default_handler
+    async def request_handler(context: PlaywrightCrawlingContext) -> None:
+        response = await context.response.text()
+        response_headers = dict(json.loads(response)).get('headers', {})
+
+        for key, val in response_headers.items():
+            headers[key] = val
+
+    await crawler.run([Request.from_url('https://httpbin.org/get', headers=custom_headers)])
+    for key, value in custom_headers.items():
+        assert headers[key] == value
```

Comment:
I believe we could compare the dicts as a whole like this:

```suggestion
    assert headers == custom_headers
```

also could we use a better names, e.g. `request_headers` and `response_headers`? :slightly_smiling_face:

Reply:
Dictionary matching doesn't work here because `set_extra_http_headers` updates headers rather than completely replacing them. So `response_headers` will contain more headers than `request_headers`.

But yes, I think it's better to do a more explicit check and update the names )